contract.py

```diff
@@ -317,7 +251,6 @@ def enforce_step(self, n: UInt64) -> None:
     ##############################################
     @subroutine
     def calculate_mab(self) -> UInt64:
```

Comment:
I'm about to make a suggestion that I don't if it is actually possible.

Right now, your `calculate_mab` function uses ALGO-specific things, like the template variables and `self`. Instead, make two functions: `calculate_mab_pure` and `calculate_mab` where `_pure` is written in pure Python and could be called outside of AlgoKit, then `calculate_mab = calculate_mab_pure(self.funding, TemplateVar[xyz], ...)`.

Now you can be positive that `simulate.py` is actually showing the same computation that the contract is using and not something different.